js/ui/components/clubhouse.js

```diff
@@ -114,12 +115,31 @@ var ClubhouseNotificationSource = new Lang.Class({
     },
 });
 
+var ClubhouseOpenButton = new Lang.Class({
+    Name: 'ClubhouseOpenButton',
+    Extends: St.Button,
+
+    _init: function(params) {
+        this.parent(params);
+
+        this._icon = new St.Icon({ style_class: 'clubhouse-open-button-icon' });
+        this.child = this._icon;
```

Comment:
I think it's more appropriate to call `this.add_child(this._icon);` here.

Reply:
Indeed, thanks.

Reply:
I remember now why I did it this way. Changing this line to use `add_child` throws an error. Here is the traceback: 

```
== Stack trace for context 0x55d402f1b650 ==

(gnome-shell:6234): Gjs-CRITICAL **: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.

(gnome-shell:6234): Gjs-CRITICAL **: The offending signal was destroy on Gjs_BaseIcon 0x55d406829cb0.
== Stack trace for context 0x55d402f1b650 ==

(gnome-shell:6234): Gjs-CRITICAL **: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.

(gnome-shell:6234): Gjs-CRITICAL **: The offending signal was actor-removed on Gjs_IconGrid 0x55d4067be600.
== Stack trace for context 0x55d402f1b650 ==

(gnome-shell:6234): Gjs-CRITICAL **: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.

(gnome-shell:6234): Gjs-CRITICAL **: The offending signal was destroy on StButton 0x55d406831620.
== Stack trace for context 0x55d402f1b650 ==
```